Sophia Sun Project Pitch

```diff
@@ -0,0 +1,244 @@
+ 
+# How did you come up with it?
+
+I came up with my idea for this project while looking at the DSL that played the Batman theme song from an Excel spreadsheet! 
+
+
+### Interface (syntax)
+# What kinds of things can the user say in your language. How, generally, do they say it?  
+
+There are two ways that I envision for a user to interact with the language. One way would be for the user to open a text file and write the name of a chord on each line. For example, if the user wanted to translate nine chords, he/she would list them like this:
+
+GDom7
+bdim65
+Ab7/EB
+Db/F
+Gb
+Db/F
+Bb7/D
+Gb 
+Bb7
+
+One thing I have to account for is that there are a lot of different ways to express the name of a chord. For example, sometimes whether a chord is major or minor is determined by whether the letter representing the root of the chord is lowercase or uppercase. Sometimes, instead of representing the inversion of a chord numerically, the chord sheet just specifies the root of the chord. These are things I need to be careful about in terms of parsing. 
+
+The user would be able to say the root of the chord, the quality of the chord, and the inversion of the chord.
+
+For the root of the chord, the user has the following options:
+A A# Ab B B# Bb C C# Cb D D# Db E Eb F F# G G# Gb 
+a a# ab b b# bb c c# cb d d# db e eb f f# g g# gb 
+
+For the quality of the chord, the user has the following options:
+Maj M Min m Dom dom Dim dim (I allow for capital / lowercase because I want to make sure that my DSL works with a variety of input types).
+
+For the inversion of the chord, the user has the following options: 
+7 65 43 2 64 63 9
+The user can also specify the inversion of the chord using a name in the chord that is the base, represented by any of the following: 
+A A# Ab B B# Bb C C# Cb D D# Db E Eb F F# G G# Gb 
+
+Generally, I expect the user to copy paste the chord names from an existing chord sheet. 
+
+Another idea that I had in the previous brainstorming assignment is for my language to provide a text input box that allows the following drop down boxes:
+root of the chord
+quality of the chord
+inversion of the chord
+base note of the chord
+
+However, I chose not to go through with this idea. Although the result would be easier to parse (the input format separates different elements that I would otherwise have to parse through a string representing many aspects of the entire chord), I think creating the interface would be a lot of work. Also, a user might not want to spend time translating existing chords into my program, especially if this would require some knowledge about parsing the chords. This is also very tedious and time consuming.
```

Comment:
I agree with you here. I think one of the advantages of having a DSL is that you can type in your input, which should be quick, instead of interfacing with drop down menus, which would take longer. Personally, I prefer if I can continuously use the keyboard or continuously use the mouse instead of having to switch between them, which is something a dropdown menu may require.